client/me/help/help-happiness-engineers/index.jsx

```diff
@@ -1,57 +1,49 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	PureRenderMixin = require( 'react-pure-render/mixin' );
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import React from 'react';
+import { shuffle } from 'lodash';
 
 /**
  * Internal dependencies
  */
-var HappinessEngineersStore = require( 'lib/happiness-engineers/store' ),
-	HappinessEngineersActions = require( 'lib/happiness-engineers/actions' ),
-	FormSectionHeading = require( 'components/forms/form-section-heading' ),
-	Gravatar = require( 'components/gravatar' );
-
-module.exports = React.createClass( {
-	displayName: 'HelpHappinessEngineers',
-
-	mixins: [ PureRenderMixin ],
-
-	componentDidMount: function() {
-		HappinessEngineersStore.on( 'change', this.refreshHappinessEngineers );
-		HappinessEngineersActions.fetch();
-	},
-
-	componentWillUnmount: function() {
-		HappinessEngineersStore.removeListener( 'change', this.refreshHappinessEngineers );
-	},
-
-	getInitialState: function() {
-		return {
-			happinessEngineers: HappinessEngineersStore.getHappinessEngineers()
-		};
-	},
-
-	refreshHappinessEngineers: function() {
-		this.setState( { happinessEngineers: HappinessEngineersStore.getHappinessEngineers() } );
-	},
-
-	render: function() {
-		if ( this.state.happinessEngineers.length === 0 ) {
-			return null;
-		}
-		return (
-			<div className="help-happiness-engineers">
-				{ this.translate( '{{headline}}We care about your happiness!{{/headline}}{{p}}They don\'t call us Happiness Engineers for nothing. If you need help, we\'re here for you!{{/p}}', {
+import FormSectionHeading from 'components/forms/form-section-heading';
+import Gravatar from 'components/gravatar';
+import QueryHappinessEngineers from 'components/data/query-happiness-engineers';
+import { getHappinessEngineers } from 'state/happiness-engineers/selectors';
+
+function HelpHappinessEngineers( props ) {
+	return (
+		<div className="help-happiness-engineers">
+			{ props.translate( '{{headline}}We care about your happiness!{{/headline}}' +
+				'{{p}}They don\'t call us Happiness Engineers for nothing. ' +
+				'If you need help, we\'re here for you!{{/p}}',
+				{
 					components: {
 						headline: <FormSectionHeading />,
 						p: <p className="help-happiness-engineers__description" />
 					}
-				} ) }
-				<div className="help-happiness-engineers__tray">
-					{ this.state.happinessEngineers.map( happinessEngineer => <Gravatar key={ happinessEngineer.avatar_URL } user={ { avatar_URL: happinessEngineer.avatar_URL } } size={ 42 } /> ) }
-				</div>
+				}
+				) }
+			<div className="help-happiness-engineers__tray">
+				{ shuffle( props.happinessEngineers ).map(
+					happinessEngineer => <Gravatar
+						key={ happinessEngineer }
+						user={ { avatar_URL: happinessEngineer } }
+						size={ 42 } /> )
+				}
 			</div>
-		);
+			{ props.happinessEngineers.length === 0 && <QueryHappinessEngineers /> }
```

Comment:
I know it'll never happen, but in other state, this would be a good example of why `null` as an initial value for the reducer may be wise, since there's no way to distinguish here between having received _anything_ and having received an _empty array_. If the initial reducer state was `null`, you could be certain that if the state was `null`, then nothing has been received. This could even be abstracted to a selector like `hasReceivedHappinessEngineers = ( state ) => !! state.happinessEngineers.items;`